pkgs/development/tools/godot/mono/default.nix

```diff
@@ -0,0 +1,94 @@
+{ godot
+, lib
+, callPackage
+, mkNugetDeps
+, mkNugetSource
+, mono
+, dotnet-sdk
+, writeText
+}:
+
+godot.overrideAttrs (self: base: {
+  pname = "godot-mono";
+
+  passthru = let bp = base.passthru; in bp // {
+    sconsFlagSet = bp.sconsFlagSet // {
+      module_mono_enabled = true;
+      mono_prefix = mono;
+    };
+
+    buildDescription = "mono build";
+
+    glue = callPackage ./glue.nix {};
+
+    make-deps = callPackage ./make-deps.nix {};
+  };
+
+  nativeBuildInputs = base.nativeBuildInputs ++ [ mono dotnet-sdk ];
```

Comment:
This should be done via an extraBuildInputs input.

Reply:
so instead of:
```nix
nativeBuildInputs = base.nativeBuildInputs ++ [ mono dotnet-sdk ];
```
do:
```nix
extraNativeBuildInputs = [ mono dotnet-sdk ];
```
?

I wonder why this property exists, given that nativeBuildInputs can simply be overridden...  And is there a reason to prefer the "extra" variant?

Reply:
I was more thinking about something like this:

```nix
{
...
, extraBuildInputs ? [ ]
}:

stdenv.mkDerivation {
...
  nativeBuildInputs = [ something ]
    ++ extraBuildInputs;
...
};

Reply:
Oh, I thought you were referring to the standard `mkDerivation` attribute called `extraBuildInputs`.

The parameters to the derivation, like your `extraBuildInputs` one, are passed in where it is called from `top-level/all-packages.nix`.  Basically `all-packages.nix` would have a line like this:

```nix
{
  # ...
  godot-mono = callPackage ../development/tools/godot/mono { extraBuildInputs = [ mono dotnet-sdk ]; };
  # ...
}
```

Why would you want that?